stripe/src/main/java/com/stripe/android/view/CardInputView.java

```diff
@@ -40,9 +55,118 @@ private void initView() {
         mExpiryDateEditText = (EditText) findViewById(R.id.et_expiry_date);
         mCvcNumberEditText = (EditText) findViewById(R.id.et_cvc_number);
-        mCardNumberSpace = (Space) findViewById(R.id.space_in_container);
+        mCardNumberSpace = findViewById(R.id.space_in_container);
+        mCardNumberIsViewed = true;
+
+        mCardNumberEditText.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
+                    scrollLeft();
+                }
+            }
+        });
+
+        mExpiryDateEditText.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
+                    scrollRight();
+                }
+            }
+        });
+
+        mCvcNumberEditText.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
+                    scrollRight();
+                }
+            }
+        });
+
+        mExpiryDateEditText.setOnKeyListener(new OnKeyListener() {
+            @Override
+            public boolean onKey(View v, int keyCode, KeyEvent event) {
+                if (keyCode == KeyEvent.KEYCODE_DEL
+                        && event.getAction() == KeyEvent.ACTION_DOWN) {
+                    if (mExpiryDateEditText.getText().length() == 0) {
+                        mCardNumberEditText.requestFocus();
+                    }
+                }
+                return false;
+            }
+        });
+
+        mCardNumberEditText.setCardNumberCompleteListener(
+                new CardNumberEditText.CardNumberCompleteListener() {
+                    @Override
+                    public void onCardNumberComplete() {
+                        mExpiryDateEditText.requestFocus();
+                        scrollRight();
+                    }
+                });
+    }
+
+    @Override
+    protected void onAttachedToWindow() {
+        super.onAttachedToWindow();
+
+    }
+
+    @Override
+    protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
+        super.onLayout(changed, left, top, right, bottom);
+        // We set this length the first time the control is laid out, because the
+        // value for the actual width of mScrollView prior to this will be zero.
+        if (!mInitializedFlag) {
+            mInitializedFlag = true;
+            mScrollViewWidth = mScrollView.getMeasuredWidth();
+            mScrollToPostion = mScrollViewWidth;
+            ViewGroup.LayoutParams widthParams = mCardNumberSpace.getLayoutParams();
+            widthParams.width = mScrollViewWidth;
+            mCardNumberSpace.setLayoutParams(widthParams);
+        }
+    }
+
+    private void scrollRight() {
```

Comment:
@mrmcduff-stripe Some comments as to why the expiry_date and cvc_number become `VISIBLE` on scrolling right and why they are `INVISIBLE` when scrolling left would be helpful.

Also, a screenshot from the emulator or the phone would really help understand how the layout looks on the device.